IaC/AKS/cluster/manifests/cert-manager/clusterissuer.yaml

```diff
@@ -0,0 +1,33 @@
+# use staging letsencrypt for testing. reduce chances of hitting rate limit
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-staging
+spec:
+  acme:
+    email: ${Ngsa_Email}
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
+    privateKeySecretRef:
+      name: letsencrypt-staging-account-key
+    solvers:
+    - http01:
+        ingress:
+          class: istio
+
+---
+
+# use production letsencrypt for real certs
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+spec:
+  acme:
+    email: ${Ngsa_Email}
```

Comment:
do we need `sed` to replace ${Ngsa_Email} with the environment variable? or kubectl automatically picks up the environment variable value get this yaml gets applied?

Reply:
`envsubst` is being used to update the file before sending it to kubectl.
ex: `envsubst < clusterissuer.yaml | kubectl apply -f -`

Another idea was to have an example file like `clusterissuer.yaml.example`. This would have the templated values.
Then the command would be
```bash
envsubst < clusterissuer.yaml.example > clusterissuer.yaml
kubectl apply -f clusterissuer.yaml
```
clusterissuer.yaml could then be added to gitignore to prevent accidental checkins. This is the main reason. Thoughts?